src/lib/request.js

```diff
@@ -32,102 +56,71 @@ export default function (relativeUrl, opts, callback) {
-                if (typeof(content) === 'object') {
-                    const keys = Object.keys(content);
-                    let key;
-
-                    for (key of keys) {
-                        const cleanKey = key.replace(/^components\//, '');
-
-                        content[cleanKey] = content[key];
-                        delete(content[key]);
-                    }
-                }
-
-                // If using "sections", morph the content into the arbitrary keys => content object.
-                if (usingSections) {
-                    const templateVariableNames = Object.keys(requestedTemplate);
-                    let templateVariable;
-                    for (templateVariable of templateVariableNames) {
-                        content[templateVariable] = content[requestedTemplate[templateVariable]];
-                        delete content[requestedTemplate[templateVariable]];
-                    }
+        headers,
+        credentials: 'include',
+        body: !options.requestOptions.formData ? stringify(data) : data,
+    }).then(response => {
+        if (response.headers.get('content-type') === 'application/json') {
+            return response.json();
+        }
+        return response.text();
+    }).then(response => {
+        const content = options.remote ? response.content : response;
+        let ret;
+
+        if (usingTemplates) {
+            // Remove the `components` prefix from the response if it's an object
+            if (typeof(content) === 'object') {
+                const keys = Object.keys(content);
+                let key;
+
+                for (key of keys) {
+                    const cleanKey = key.replace(/^components\//, '');
+
+                    content[cleanKey] = content[key];
+                    delete(content[key]);
                 }
+            }
 
-                if (options.remote) {
-                    ret = response;
-                    ret.content = content;
-                } else {
-                    ret = content;
+            // If using "sections", morph the content into the arbitrary keys => content object.
+            if (usingSections) {
+                const requestedTemplate = options.requestOptions.template;
+                const templateVariableNames = Object.keys(requestedTemplate);
+                let templateVariable;
+                for (templateVariable of templateVariableNames) {
+                    content[templateVariable] = content[requestedTemplate[templateVariable]];
+                    delete content[requestedTemplate[templateVariable]];
                 }
-            } else {
-                ret = response;
             }
 
-            callback(null, ret);
-        },
-        error: (XHR, textStatus, err) => {
-            callback(err);
-        },
-        data,
-        headers,
-    });
+            if (!options.remote) {
+                ret = content;
+            }
```

Comment:
Why did we stop returning values for cases when `options.remote == true`?
Seems like now for cases when `usingTemplates == true && options.remote == true` we are gonna return `undefined` instead of `response`

Reply:
Fixed that